es74x-cloud/src/main/scala/tech/beshu/ror/proxy/es/clients/HighLevelClientBasedClusterAdminClient.scala

```diff
@@ -0,0 +1,279 @@
+import org.elasticsearch.action.admin.cluster.repositories.put.{PutRepositoryRequest, PutRepositoryRequestBuilder}
+import org.elasticsearch.action.admin.cluster.repositories.verify.{VerifyRepositoryRequest, VerifyRepositoryRequestBuilder, VerifyRepositoryResponse}
+import org.elasticsearch.action.admin.cluster.reroute.{ClusterRerouteRequest, ClusterRerouteRequestBuilder, ClusterRerouteResponse}
+import org.elasticsearch.action.admin.cluster.settings.{ClusterUpdateSettingsRequest, ClusterUpdateSettingsRequestBuilder, ClusterUpdateSettingsResponse}
+import org.elasticsearch.action.admin.cluster.shards.{ClusterSearchShardsRequest, ClusterSearchShardsRequestBuilder, ClusterSearchShardsResponse}
+import org.elasticsearch.action.admin.cluster.snapshots.create.{CreateSnapshotRequest, CreateSnapshotRequestBuilder, CreateSnapshotResponse}
+import org.elasticsearch.action.admin.cluster.snapshots.delete.{DeleteSnapshotRequest, DeleteSnapshotRequestBuilder}
+import org.elasticsearch.action.admin.cluster.snapshots.get.{GetSnapshotsRequest, GetSnapshotsRequestBuilder, GetSnapshotsResponse}
+import org.elasticsearch.action.admin.cluster.snapshots.restore.{RestoreSnapshotRequest, RestoreSnapshotRequestBuilder, RestoreSnapshotResponse}
+import org.elasticsearch.action.admin.cluster.snapshots.status.{SnapshotsStatusRequest, SnapshotsStatusRequestBuilder, SnapshotsStatusResponse}
+import org.elasticsearch.action.admin.cluster.state.{ClusterStateRequest, ClusterStateRequestBuilder, ClusterStateResponse}
+import org.elasticsearch.action.admin.cluster.stats.{ClusterStatsRequest, ClusterStatsRequestBuilder, ClusterStatsResponse}
+import org.elasticsearch.action.admin.cluster.storedscripts.{DeleteStoredScriptRequest, DeleteStoredScriptRequestBuilder, GetStoredScriptRequest, GetStoredScriptRequestBuilder, GetStoredScriptResponse, PutStoredScriptRequest, PutStoredScriptRequestBuilder}
+import org.elasticsearch.action.admin.cluster.tasks.{PendingClusterTasksRequest, PendingClusterTasksRequestBuilder, PendingClusterTasksResponse}
+import org.elasticsearch.action.ingest.{DeletePipelineRequest, DeletePipelineRequestBuilder, GetPipelineRequest, GetPipelineRequestBuilder, GetPipelineResponse, PutPipelineRequest, PutPipelineRequestBuilder, SimulatePipelineRequest, SimulatePipelineRequestBuilder, SimulatePipelineResponse}
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.ClusterAdminClient
+import org.elasticsearch.cluster.{ClusterName, ClusterState}
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.tasks.TaskId
+import org.elasticsearch.threadpool.ThreadPool
+import tech.beshu.ror.proxy.es.exceptions.NotDefinedForRorProxy
+
+class HighLevelClientBasedClusterAdminClient(esClient: RestHighLevelClientAdapter)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Number of methods in class exceeds 30](https://app.codacy.com/app/scarduzio/elasticsearch-readonlyrest-plugin/pullRequest?prid=4591272)